src/test/javascript/spec/component/tutor-dashboard/tutor-exercise-dashboard.spec.ts

```diff
@@ -0,0 +1,111 @@
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('TutorExerciseDashboardComponent', () => {
+    let comp: TutorExerciseDashboardComponent;
+    let fixture: ComponentFixture<TutorExerciseDashboardComponent>;
+    let modelingSubmissionService: ModelingSubmissionService;
+
+    let modelingSubmissionStub: SinonStub;
+
+    const exercise = { id: 20, type: ExerciseType.MODELING, tutorParticipations: [{ status: TutorParticipationStatus.TRAINED }] } as ModelingExercise;
+    const submission = { id: 30 } as ModelingSubmission;
+    const lockLimitErrorResponse = new HttpErrorResponse({ error: { errorKey: 'lockedSubmissionsLimitReached' } });
+
+    beforeEach(async () => {
+        return TestBed.configureTestingModule({
+            imports: [ArTEMiSTestModule, ArTEMiSSharedModule, ArTEMiSTutorCourseDashboardModule, RouterModule],
+            declarations: [
+                TutorExerciseDashboardComponent,
+                MockComponent(HeaderExercisePageWithDetailsComponent),
+                MockComponent(SidePanelComponent),
+                MockComponent(ModelingEditorComponent),
+                MockComponent(ResultComponent),
+                MockComponent(AssessmentInstructionsComponent),
+            ],
+            providers: [
+                JhiLanguageHelper,
+                { provide: JhiAlertService, useClass: MockAlertService },
+                { provide: ActivatedRoute, useClass: MockActivatedRoute },
+                { provide: Router, useClass: MockRouter },
+                {
+                    provide: ExerciseService,
+                    useValue: {
+                        getForTutors() {
+                            return {
+                                subscribe: (fn: (value: any) => void) =>
+                                    fn({
+                                        body: exercise,
+                                    }),
+                            };
+                        },
+                        getStatsForTutors() {
+                            return of();
+                        },
+                    },
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(TutorExerciseDashboardComponent);
+                comp = fixture.componentInstance;
+                modelingSubmissionService = TestBed.get(ModelingSubmissionService);
+
+                comp.exerciseId = exercise.id;
+
+                modelingSubmissionStub = stub(modelingSubmissionService, 'getModelingSubmissionForExerciseWithoutAssessment');
+            });
+    });
+
+    afterEach(() => {
+        modelingSubmissionStub.restore();
+    });
+
+    it('should set unassessedSubmission if lock limit is not reached', () => {
+        modelingSubmissionStub.returns(of(submission));
+
+        comp.loadAll();
+
+        expect(modelingSubmissionStub).to.have.been.calledOnceWithExactly(exercise.id);
```

Comment:
I think it's a good practice to both verify that a stub was called and also check which arguments where used.